src/WebJobs.Extensions.DurableTask/EndToEndTraceHelper.cs

```diff
@@ -54,6 +59,41 @@ public static string LocalSlotName
 
 #pragma warning disable SA1117 // Parameters should be on same line or separate lines
 
+        private void SanitizeString(string? rawPayload, out string iloggerString, out string durableKustoTableString, bool isReplay = false)
+        {
+            if (isReplay)
+            {
+                iloggerString = rawPayload ?? "(replay)";
+                durableKustoTableString = "(replay)";
+            }
+
+            string payload = rawPayload ?? "(null)";
+            int numCharacters = payload.Length;
+            string sanitizedPayload = $"(Redacted {numCharacters} characters)";
+
+            iloggerString = this.shouldTraceRawData ? payload : sanitizedPayload;
+            durableKustoTableString = sanitizedPayload;
+        }
+
+        private void SanitizeException(Exception? exception, out string iloggerExceptionString, out string durableKustoTableString, bool isReplay = false)
+        {
+            string exceptionString = exception != null ? exception.ToString() : string.Empty;
+            if (exception is OrchestrationFailureException orchestrationFailureException)
+            {
+                exceptionString = orchestrationFailureException.Details;
+            }
+
+            string sanitizedString = exception != null ? $"{exception.GetType().FullName}\n{exception.StackTrace}" : string.Empty;
+            iloggerExceptionString = this.shouldTraceRawData ? exceptionString : sanitizedString;
+
+            if (isReplay)
+            {
+                durableKustoTableString = "(replay)";
```

Comment:
Why do we assign `durableKustoTableString` here if we are reassigning it on line 94 (if `isReplay` is true)?

Reply:
Answered here: https://github.com/Azure/azure-functions-durable-extension/pull/2844#discussion_r1663332537 . Let's look to continue the discussion _there._